File: app/src/main/java/com/example/kotlindemo/di/local/NoteDao.java
package com.example.kotlindemo.di.local;

import static androidx.room.OnConflictStrategy.REPLACE;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface NoteDao {

    @Insert(onConflict = REPLACE)
    void insert(NoteEntity data);


    @Query("SELECT * FROM note_entity ")
    List<NoteEntity> getNoteList();
}

File: app/src/main/java/com/example/kotlindemo/notes_app/data/db/dao/NoteDao.kt
package com.example.kotlindemo.notes_app.data.db.dao

import androidx.room.*
import com.example.kotlindemo.notes_app.data.model.NoteEntity
import io.reactivex.Observable

@Dao
interface NoteDao {

    @Query("SELECT * FROM notes_table ORDER BY local_id DESC")
    fun getNotesList(): Observable<List<NoteEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(notesEntity: NoteEntity?)

    @Query("delete from notes_table")
    fun deleteAllNotes()

    @Update
    suspend fun updateNote(notesEntity: NoteEntity?)
}